pgx/src/datum/into.rs

```diff
@@ -32,6 +33,60 @@ pub trait IntoDatum {
     fn array_type_oid() -> pg_sys::Oid {
         unsafe { pg_sys::get_array_type(Self::type_oid()) }
     }
+
+    /// Is a Datum of this type compatible with another Postgres type?
+    ///
+    /// An example of this are the Postgres `text` and `varchar` types, which are both
+    /// technically compatible from a Rust type perspective.  They're both represented in Rust as
+    /// `String` (or `&str`), but the underlying Postgres types are different.
+    ///
+    /// If implementing this yourself, you likely want to follow a pattern like this:
+    ///
+    /// ```text
```

Comment:
```suggestion
    /// ```rust,no_run
```

Reply:
No, because it's not exactly valid Rust code and I think the docs system will still try to ensure it parses.  I'll take a look at making it actually valid rust.

Reply:
If all the type and imports are correct, sure it is. This builds just fine ([Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=cf6526e9cd236a2d51a8ab1fbd420287)):
```rust
pub mod pg_sys {
    #[derive(PartialEq)]
    pub struct Oid();
    pub const TEXTOID: Oid = Oid();
}

struct Something;
impl Something {
    fn type_oid() -> pg_sys::Oid {
        pg_sys::Oid()
    }

    fn is_compatible_with(other: pg_sys::Oid) -> bool {
        // first, if our type is the other type, then we're compatible
        Self::type_oid() == other

     // and here's the other type we're compatible with
     || other == pg_sys::TEXTOID
    }
}
```

Reply:
Yah, I got something hacked that's now `rust,no_run`.